File: server/src/main/java/edu/brown/cs/student/main/Message/MessageType.java
package edu.brown.cs.student.main.Message;

/**
 * MessageType enum to represent all message communication types (including those between the server
 * and client)
 */
public enum MessageType {
  NEW_CLIENT_NO_CODE,
  NEW_CLIENT_WITH_CODE,
  UPDATE_BOARD,
  CURRENT_BOARD,
  SET_GAME_CODE,
  ERROR,
  SUCCESS,
  JOIN_ERROR,
  RESTART_GAME,
  JOIN_SUCCESS
}

File: client/src/components/message/MessageType.ts
/**
 * A enum for the possible messages sent and received to the our Minesweeper game
 * server via a websocket.
 */
enum MessageType {
    NEW_CLIENT_NO_CODE = "NEW_CLIENT_NO_CODE",
    NEW_CLIENT_WITH_CODE = "NEW_CLIENT_WITH_CODE",
    UPDATE_BOARD = "UPDATE_BOARD",
    CURRENT_BOARD = "CURRENT_BOARD",
    SET_GAME_CODE = "SET_GAME_CODE",
    ERROR = "ERROR",
    SUCCESS = "SUCCESS",
    JOIN_ERROR = "JOIN_ERROR",
    RESTART_GAME = "RESTART_GAME",
    JOIN_SUCCESS = "JOIN_SUCCESS",
  }
  
  export default MessageType;